middlewares/checkAuth.js

```diff
@@ -1,15 +1,36 @@
 import passport from "passport";
 import constants from "../helpers/constants";
+import models from "../models";
 
-const { UNAUTHORIZED } = constants.statusCode;
+const { Permission } = models;
+const { UNAUTHORIZED, FORBIDDEN } = constants.statusCode;
 export default (req, res, next) => {
-  passport.authenticate("jwt", { session: false }, (err, user) => {
+  passport.authenticate("jwt", { session: false }, async (err, user) => {
+    const { originalUrl } = req;
+    const [, , , resource] = originalUrl.split("/");
     if (!user) {
       return res.status(UNAUTHORIZED).json({
         message: "Please provide a token to perform this action"
       });
     }
+    const { roleId } = user;
+    const permissionsList = await Permission.findAll({
+      where: { roleId },
+      raw: true
+    });
     req.user = user.dataValues;
-    next();
+    let allow = false;
+    permissionsList.forEach(function(perms) {
+      if (perms.resource === resource) {
+        if (req.method === "POST" && perms.create_) allow = true;
+        else if (req.method === "GET" && perms.read_) allow = true;
+        else if (req.method === "PUT" && perms.update_) allow = true;
+        else if (req.method === "DELETE" && perms.delete_) allow = true;
+      }
+    });
+    if (resource === "roles" && req.method === "POST") allow = true;
+    if (resource === "users" && req.method === "PUT") allow = true;
+    if (allow) next();
+    else res.status(FORBIDDEN).send({ message: "Access Denied" });
```

Comment:
@niomwungeri-fabrice, can you please use a user-friendly error message here? 
You can check in the resource we shared in our slack channel.